### PR TITLE
12683: add extra conditional questions to proposed actions on landuse-form

### PR DESCRIPTION
**Pull request by trbmcginnis**
**Big Picture Summary**
This PR adds several conditional questions to the Proposed Actions section of the Land Use Form.

**Which major feature does this fit into?**
Land Use Form

**Technical Explanation — How does it work?**
The `proposed-action-editor` component was updated to include these extra conditional questions. 

Addresses [AB#12683](https://dcp-paperless.visualstudio.com/d36fd830-9029-4b77-b0c4-b0df2012eb98/_workitems/edit/12683)

### CODE REVIEW

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {{#let @landuseActionForm as |landuseActionForm|}}
   <fieldset class="fieldset">
+    
+    {{#if landuseActionForm.data.dcpDatadifferentiationmultipleactionsametype}}
+      <p>{{landuseActionForm.data.dcpDatadifferentiationmultipleactionsametype}}. This information differentiates multiple Actions with the same Action Code.</p>
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.hbs:5]: Cool. That's weird that this part is in the wireframes but not the users story.

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs [hunk 2]
```diff
@@ -15,9 +20,163 @@
       </span>
     </Ui::Legend>
 
+    <Ui::Question
+      class="no-margin-top"
+      @required={{true}}
+      data-test-dcpapplicantispublicagencyactions-radio-group
+      as |dcpApplicantispublicagencyactionsQ|
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.hbs:27]: Nice, thanks!

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs [hunk 3]
```diff
@@ -15,9 +20,163 @@
       </span>
     </Ui::Legend>
 
+    <Ui::Question
+      class="no-margin-top"
+      @required={{true}}
+      data-test-dcpapplicantispublicagencyactions-radio-group
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.hbs:26]: Actually no need to add the test selector here. The beauty of the Field component is that it generates these test selectors for us

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs [hunk 4]
```diff
@@ -15,9 +20,163 @@
...
+      <dcpApplicantispublicagencyactionsQ.Legend>
+        Is this an action where the applicant is solely a public agency?
+      </dcpApplicantispublicagencyactionsQ.Legend>
+
+      <landuseActionForm.Field
+        @attribute="dcpApplicantispublicagencyactions"
+        @type="radio-group"
+      as |DcpApplicantispublicagencyactionsRadioGroup|>
+        <DcpApplicantispublicagencyactionsRadioGroup
+          @options={{optionset 'landuseAction' 'dcpApplicantispublicagencyactions' 'list'}}
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.hbs:38]: yay, thanks!

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs [hunk 5]
```diff
@@ -15,9 +20,163 @@
...
+      @required={{true}}
+      data-test-dcpapplicantispublicagencyactions-radio-group
+      as |dcpApplicantispublicagencyactionsQ|
+    >
+      <dcpApplicantispublicagencyactionsQ.Legend>
+        Is this an action where the applicant is solely a public agency?
+      </dcpApplicantispublicagencyactionsQ.Legend>
+
+      <landuseActionForm.Field
+        @attribute="dcpApplicantispublicagencyactions"
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.hbs:34]: weird, how come I don't see this field in AB#12683?

File: client/app/components/packages/landuse-form/proposed-action-editor.js [hunk 6]
```diff
@@ -1,13 +1,36 @@
...
       return optionset(['landuseAction', 'dcpPreviouslyapprovedactioncode', 'label', previouslyApprovedActionCode]);
     } return '-- select an action --';
   }
+
+  @action
+  setPreviouslyApprovedActionCodeFields(actionCode) {
+    if (actionCode) {
+      this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode = actionCode;
+      this.args.landuseActionForm.data.dcpFollowuptopreviousaction = true;
+    } else if (!actionCode && !this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode) {
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.js:32]: I think `!actionCode` here is redundant, due to else if?

File: client/app/components/packages/landuse-form/proposed-action-editor.js [hunk 7]
```diff
@@ -1,13 +1,36 @@
...
     if (previouslyApprovedActionCode) {
       return optionset(['landuseAction', 'dcpPreviouslyapprovedactioncode', 'label', previouslyApprovedActionCode]);
     } return '-- select an action --';
   }
+
+  @action
+  setPreviouslyApprovedActionCodeFields(actionCode) {
+    if (actionCode) {
+      this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode = actionCode;
+      this.args.landuseActionForm.data.dcpFollowuptopreviousaction = true;
```
Comment (godfreyyeung) [inline client/app/components/packages/landuse-form/proposed-action-editor.js:31]: So this action was created to make sure that  `dcpFollowuptopreviousaction` is also set when user makes a dropdown selection? What is the purpose of `dcpFollowuptopreviousaction`?

File: client/app/helpers/optionset.js [hunk 8]
```diff
@@ -129,6 +130,10 @@ const OPTIONSET_LOOKUP = {
   },
   landuseAction: {
     dcpApplicantispublicagencyactions: YES_NO,
+    dcpIstheactiontoauthorizeorpermitanopenuse: YES_NO,
+    dcpIstheactiontoauthorizeacommercial: YES_NO,
+    dcpIndicatewhetheractionisamodification: LANDUSE_ACTION_OPTIONSETS.DCPINDICATEWHETHERACTIONISAMODIFICATION,
```
Comment (godfreyyeung) [inline client/app/helpers/optionset.js:135]: Thanks!!

**Review by godfreyyeung — APPROVED** [review]
Ohh mann.. These forms got me dizzy 😵  I can't say I understand the Land Use form business logic but love the code quality and use of optionsets!

File: client/app/components/packages/landuse-form/proposed-action-editor.js (hunk 7 above)
Comment (trbmcginnis) [reply client/app/components/packages/landuse-form/proposed-action-editor.js:31]: Yes it was -- `dcpFollowuptopreviousaction` is required for display logic -- not sure it's totallyyy necessary (like we could just check that `dcpPreviouslyapprovedactioncode` is not a blank string or undefined but `dcpFollowuptopreviousaction` was included in the wireframe so I went ahead and incldued it here

File: client/app/components/packages/landuse-form/proposed-action-editor.hbs (hunk 5 above)
Comment (trbmcginnis) [reply client/app/components/packages/landuse-form/proposed-action-editor.hbs:34]: I think it was included in other PRs so they didn't include it in this task

**Pull request merged**